include/pybind11/detail/common.h

```diff
@@ -205,6 +205,7 @@ extern "C" {
 #define PYBIND11_TRY_NEXT_OVERLOAD ((PyObject *) 1) // special failure return code
 #define PYBIND11_STRINGIFY(x) #x
 #define PYBIND11_TOSTRING(x) PYBIND11_STRINGIFY(x)
+#define PYBIND11_STRCONCAT(first, second) first##second
```

Comment:
Minor nitpick: this isn't actually string concatenation; perhaps rename to just `PYBIND11_CONCAT`?